testsuite/tests/ui/master/test_tenant.py

```diff
@@ -0,0 +1,118 @@
+"""
+    Tests that test the tenant modification such as creation, deletion, edit or resume.
+"""
+import pytest
+
+from testsuite.ui.views.admin.login import LoginView
+from testsuite.ui.views.common.foundation import NotFoundView
+from testsuite.utils import blame
+from testsuite.ui.utils import assert_displayed_in_new_tab
+from testsuite.ui.views.master.audience.tenant import TenantDetailView, TenantEditView
+from testsuite.ui.views.devel import LandingView
+
+
+@pytest.fixture(scope="module")
+def custom_blame(request):
+    """ creates a simple blame string """
+    return blame(request, "org-name")
+
+
+@pytest.fixture(scope="module")
+def ui_tenant(custom_blame, custom_ui_tenant):
+    """ overrides the ui_tenant fixture and creates and returns new tenant """
+    return custom_ui_tenant(username=custom_blame, email=custom_blame, password="12345678", organisation=custom_blame)
+
+
+def test_create_tenant(ui_tenant, custom_blame, navigator, browser, master_threescale):
+    """
+    Test:
+        - Creates tenant via UI
+        - Checks whether it exists
+    """
+    account_id = ui_tenant.entity['signup']['account']['id']
+    account = master_threescale.accounts.read(account_id)
+
+    assert account.entity_name == custom_blame
+
+    detail_view = navigator.navigate(TenantDetailView, account=account)
+
+    assert detail_view.public_domain.text == account.entity['domain']
+    assert detail_view.admin_domain.text == account.entity['admin_domain']
+
+    assert_displayed_in_new_tab(browser, detail_view.open_public_domain, LandingView)
+    assert_displayed_in_new_tab(browser, detail_view.open_admin_domain, LoginView)
+
+
+# pylint: disable=unused-argument
+def test_edit_tenant(master_login, navigator, custom_tenant, master_threescale):
+    """
+    Test:
+        - Create tenant via API
+        - Edit tenant via UI
+        - check whether it was edited
+    """
+    tenant = custom_tenant()
+    account_id = tenant.entity['signup']['account']['id']
+    account = master_threescale.accounts.read(account_id)
+
+    old_name = account.entity_name
+    edit = navigator.navigate(TenantEditView, account=account)
+
+    edit.update(org_name="updated_name")
+    account = master_threescale.accounts.read(account_id)
+
+    assert account.entity_name == "updated_name"
+    assert account.entity_name != old_name
```

Comment:
I agree, the second assert was added after the @jsmolar pointed out that I am not actually checking if there was a change and that the first one is just checking whether the name is set to some constant. I think that this additional assert just adds a secondary check in case that the accounts name is actually the same as we are changing it to (Undefined behavior?)

If you think that one of them should be removed, which one do you recommend ?

Reply:
As Laso wrote. The second assertion is to verify if the name was really changed.

Reply:
I am not against, double-checking being better than nothing but I cannot imagine any possible scenario where `account.entity_name == "updated_name"`  check will pass but second check `assert account.entity_name != old_name` will fail

Reply:
before - `updated_name`
after - `updated_name`

Altho this is imho the "Undefined behavior" :/

Reply:
You are always creating a new tenant that has a unique generated name.

Reply:
How about just switching the order of the asserts? First if it changed and second if it changes to the correct value?

Reply:
I like the @pehala 's idea ... what do you think @dhlavac ?